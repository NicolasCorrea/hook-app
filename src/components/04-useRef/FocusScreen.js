import { useRef } from "react"
import "../02-useEffect/effects.css"
export const FocusScreen = () => {

  const inputRef = useRef();

  const handleClick = () => {
    inputRef.current.focus();
    console.log(inputRef);
  }

  return (
    <>
      <h1>Focus Screen</h1>
      <hr />
      <input
        ref={inputRef}
        type="text"
        className="form-control"
        name="name"
        id="name"
        placeholder="Your Name"
      />

      <button
        className="btn btn-primary mt-3"
        onClick={handleClick}
      >
        Focus
      </button>

    </>
  )
}
