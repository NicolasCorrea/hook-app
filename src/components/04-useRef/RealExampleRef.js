import { useState } from "react"
import "../02-useEffect/effects.css"
import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks"
export const RealExampleRef = () => {

  const [show, setShow] = useState(false)

  return (
    <>
      <h1>Real Example Ref</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() => {
          setShow(!show);
        }}
      >
        Show/Hide
      </button>

      {show && <MultipleCustomHooks />}

    </>
  )
}
