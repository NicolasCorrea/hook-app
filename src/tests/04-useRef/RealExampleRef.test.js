import { shallow } from 'enzyme'
import { RealExampleRef } from '../../components/04-useRef/RealExampleRef'

describe('test on <RealExampleRef />', () => {
  
  test('should render', () => {
    const wrapper = shallow(<RealExampleRef />)
    expect(wrapper).toMatchSnapshot();
  })
  
  test('should show <MultipleCustomHooks />', () => {
    const wrapper = shallow(<RealExampleRef />)
    const button = wrapper.find('button');
    button.simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  })
  

})
