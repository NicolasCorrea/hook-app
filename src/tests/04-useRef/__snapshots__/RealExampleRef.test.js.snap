// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <RealExampleRef /> should render 1`] = `
<Fragment>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
